### PR TITLE
Fix connect to server and response parsing.

### PR DESCRIPTION
**Pull request by deltalejo**

### CODE REVIEW

**Comment by jotathebest** [comment]
As advice, I would not build String instances to process the server's response, it is better if you use arrays with primitive chars

File: src/UbiHttp.cpp [hunk 1]
```diff
@@ -134,21 +139,25 @@ double UbiHTTP::get(const char *device_label, const char *variable_label) {
     return ERROR_VALUE;
   }
 
-  uint16_t pathLength = _pathLength(device_label, variable_label);
-  char *path = (char *)malloc(sizeof(char) * pathLength + 1);
+  int pathLength = snprintf(NULL, 0, "/api/v1.6/devices/%s/%s/lv", device_label, variable_label);
```
Comment (crisap94) [inline src/UbiHttp.cpp:142]: I would prefer to just replace this line where _pathLength is defined as it is already being reused, also the implementation already has a description.

File: src/UbiHttp.cpp [hunk 2]
```diff
@@ -134,21 +139,25 @@ double UbiHTTP::get(const char *device_label, const char *variable_label) {
...
   sprintf(path, "/api/v1.6/devices/%s/%s/lv", device_label, variable_label);
 
   if (_debug) {
     Serial.print(F("Requesting to URL: "));
     Serial.println(path);
   }
 
-  uint16_t requestLineLength = _requestLineLength(path);
-  char *message = (char *)malloc(sizeof(char) * requestLineLength + 1);
+  int requestLineLength = snprintf(NULL, 0,
```
Comment (crisap94) [inline src/UbiHttp.cpp:151]: I would prefer to just replace this line where `_requestLineLenght` is defined as it is already being reused, also the implementation already has a description.

File: src/UbiHttp.cpp (hunk 1 above)
Comment (crisap94) [reply src/UbiHttp.cpp:142]: Following the principle of KISS (Keep it Simple, Studpid), read `snprintf` probably makes sense now, but giving semantic names to specific actions may reduce the time to fully understand the code as it is natural language.

File: src/UbiHttp.cpp [hunk 3]
```diff
@@ -157,79 +166,62 @@ double UbiHTTP::get(const char *device_label, const char *variable_label) {
...
-  double value = strtof(_charValue, NULL);
-
-  free(_charLength);
-  free(_charValue);
+  /* Skip response headers */
+  if (_debug) {
+    Serial.println("Server response:");
+  }
+  while (_client_https_ubi.connected()) {
+    String line = _client_https_ubi.readStringUntil('\n');
```
Comment (crisap94) [inline src/UbiHttp.cpp:194]: Probably the ESP32 has sufficient RAM to just spend it like this, the String object if like a PACMAN of RAM, That's why the previous implementation was doing.

The final goal of this is to unify the libraries at some point, probably some of them will have different implementations, but if we can develop the library thinking about the device with the less amount of RAM available this implementation may break due overflowing.

Comment (crisap94) [reply src/UbiHttp.cpp:194]: I'd rather change this for something less RAM expensive.

File: src/UbiHttp.cpp [hunk 4]
```diff
@@ -157,79 +166,62 @@ double UbiHTTP::get(const char *device_label, const char *variable_label) {
...
+  } else {
+    if (_debug) {
+      Serial.println(F("Could not read server's response"));
+    }
+  }
+  
   return value;
 }
 
 double UbiHTTP::_parseServerAnswer() {
```
Comment (crisap94) [inline src/UbiHttp.cpp:188]: The comments below apply for the whole function where you are using the String object.

**Review by crisap94 — CHANGES_REQUESTED** [review]
Using String object may lead to undesirable behaviors for our customers that use this library.

File: src/UbiHttp.cpp (hunk 1 above)
Comment (deltalejo) [reply src/UbiHttp.cpp]: The change is reverted.

File: src/UbiHttp.cpp (hunk 2 above)
Comment (deltalejo) [reply src/UbiHttp.cpp]: The change is reverted.

File: src/UbiHttp.cpp (hunk 3 above)
Comment (deltalejo) [reply src/UbiHttp.cpp]: The library already use one instance of String class, I will keep that one.

Comment (jotathebest) [reply src/UbiHttp.cpp]: @crisap94 @deltalejo  to use a String buffer to read the server's response is a RAM waste, I think that you should not add the piece of code of line 194

**Pull request closed without merging**